src/main/java/ca/uhn/fhir/jpa/demo/HapiProperties.java

```diff
@@ -0,0 +1,211 @@
+package ca.uhn.fhir.jpa.demo;
+
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.rest.api.EncodingEnum;
+import ca.uhn.fhir.rest.server.ETagSupportEnum;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class HapiProperties {
+    public static final String SERVER_ADDRESS = "server_address";
+    public static final String DEFAULT_PRETTY_PRINT = "default_pretty_print";
+    public static final String MAX_PAGE_SIZE = "max_page_size";
+    public static final String DEFAULT_PAGE_SIZE = "default_page_size";
+    public static final String LOGGER_NAME = "logger.name";
+    public static final String LOGGER_FORMAT = "logger.format";
+    public static final String ALLOW_EXTERNAL_REFERENCES = "allow_external_references";
+    public static final String ALLOW_MULTIPLE_DELETE = "allow_multiple_delete";
+    public static final String DATASOURCE_PASSWORD = "datasource.password";
+    public static final String DATASOURCE_USERNAME = "datasource.username";
+    public static final String DATASOURCE_URL = "datasource.url";
+    public static final String DATASOURCE_DRIVER = "datasource.driver";
+    public static final String LOGGER_LOG_EXCEPTIONS = "logger.log_exceptions";
+    public static final String LOGGER_ERROR_FORMAT = "logger.error_format";
+    public static final String PERSISTENCE_UNIT_NAME = "persistence_unit_name";
+    public static final String SERVER_BASE = "server.base";
+    public static final String TEST_PORT = "test.port";
+    public static final String SERVER_NAME = "server.name";
+    public static final String SERVER_ID = "server.id";
+    private static Properties properties;
+    private static final String HAPI_PROPERTIES = "hapi.properties";
+    private static final String FHIR_VERSION = "fhir_version";
+    private static final String DEFAULT_ENCODING = "default_encoding";
+    private static final String ETAG_SUPPORT = "etag_support";
+
+    public static Properties getProperties() {
+        if (properties == null) {
+            // Load the configurable properties file
+            InputStream in = null;
+
+            try {
+                in = HapiProperties.class.getClassLoader().getResourceAsStream(HAPI_PROPERTIES);
+                HapiProperties.properties = new Properties();
+                HapiProperties.properties.load(in);
+                in.close();
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
```

Comment:
Same deal here, I'd just re-throw these.

Reply:
I've updated to have handled exceptions re-throw the exceptions as ConfigurationExceptions.